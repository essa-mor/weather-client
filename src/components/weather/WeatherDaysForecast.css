.weather_days_forecast{
	display: flex;
}