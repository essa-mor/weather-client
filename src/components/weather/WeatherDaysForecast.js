import React from 'react';
import PropTypes from 'prop-types';

import './WeatherDaysForecast.css';
import WeatherDayIndicator from './WeatherDayIndicator';

function WeatherDaysForecast({ data }) {
	const weatherDayIndicators = data.map(({ dt, day, img, maxTemperature, minTemperature }) => (<WeatherDayIndicator
		dt={dt}
		key={day}
		day={day}
		img={`http://openweathermap.org/img/w/${img}.png`}
		maxTemperature={maxTemperature}
		minTemperature={minTemperature} />));

	return (<div className="weather_days_forecast">
		{weatherDayIndicators}
	</div>);
}

WeatherDaysForecast.propTypes = {
	data: PropTypes.arrayOf(PropTypes.shape({
		dt: PropTypes.number.isRequired,
		day: PropTypes.string.isRequired,
		maxTemperature: PropTypes.number.isRequired,
		minTemperature: PropTypes.number.isRequired,
		img: PropTypes.string.isRequired
	})).isRequired
};

export default WeatherDaysForecast;