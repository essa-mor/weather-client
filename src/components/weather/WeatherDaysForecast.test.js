import React from 'react';
import renderer from 'react-test-renderer';

import WeatherDaysForecast from './WeatherDaysForecast';

const data = [
	{ day: '2018-08-23 12:00:00', maxTemperature: 29.39, minTemperature: 27.66, img: '01d' },
	{ day: '2018-08-24 12:00:00', maxTemperature: 27.14, minTemperature: 27.14, img: '01d' },
	{ day: '2018-08-25 12:00:00', maxTemperature: 26.95, minTemperature: 26.95, img: '01d' },
	{ day: '2018-08-26 12:00:00', maxTemperature: 27.01, minTemperature: 27.01, img: '02d' },
	{ day: '2018-08-27 12:00:00', maxTemperature: 26.9, minTemperature: 26.9, img: '03d' }];

describe('WeatherDaysForecast test', () => {
	it('renders correctly', () => {
		const tree = renderer.create(<WeatherDaysForecast data={data} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});