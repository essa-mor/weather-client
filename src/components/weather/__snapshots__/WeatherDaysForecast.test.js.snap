// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDaysForecast test renders correctly 1`] = `
<div
  className="weather_days_forecast"
>
  <div
    className="weather_day_indicator"
  >
    <span
      className="weather_day"
    >
      Thu
    </span>
    <img
      alt=""
      src="http://openweathermap.org/img/w/01d.png"
    />
    <div
      className="weather_temperatures"
    >
      <span
        className="weather_max_temperature"
      >
        29
        °
      </span>
      <span
        className="weather_min_temperature"
      >
        28
        °
      </span>
    </div>
  </div>
  <div
    className="weather_day_indicator"
  >
    <span
      className="weather_day"
    >
      Fri
    </span>
    <img
      alt=""
      src="http://openweathermap.org/img/w/01d.png"
    />
    <div
      className="weather_temperatures"
    >
      <span
        className="weather_max_temperature"
      >
        27
        °
      </span>
      <span
        className="weather_min_temperature"
      >
        27
        °
      </span>
    </div>
  </div>
  <div
    className="weather_day_indicator"
  >
    <span
      className="weather_day"
    >
      Sat
    </span>
    <img
      alt=""
      src="http://openweathermap.org/img/w/01d.png"
    />
    <div
      className="weather_temperatures"
    >
      <span
        className="weather_max_temperature"
      >
        27
        °
      </span>
      <span
        className="weather_min_temperature"
      >
        27
        °
      </span>
    </div>
  </div>
  <div
    className="weather_day_indicator"
  >
    <span
      className="weather_day"
    >
      Sun
    </span>
    <img
      alt=""
      src="http://openweathermap.org/img/w/02d.png"
    />
    <div
      className="weather_temperatures"
    >
      <span
        className="weather_max_temperature"
      >
        27
        °
      </span>
      <span
        className="weather_min_temperature"
      >
        27
        °
      </span>
    </div>
  </div>
  <div
    className="weather_day_indicator"
  >
    <span
      className="weather_day"
    >
      Mon
    </span>
    <img
      alt=""
      src="http://openweathermap.org/img/w/03d.png"
    />
    <div
      className="weather_temperatures"
    >
      <span
        className="weather_max_temperature"
      >
        27
        °
      </span>
      <span
        className="weather_min_temperature"
      >
        27
        °
      </span>
    </div>
  </div>
</div>
`;
